ftwhttp/request_test.go

```diff
@@ -174,72 +156,56 @@ func TestRequestHeadersSet(t *testing.T) {
 	}
 
 	req.AddHeader("X-New-Header2", "Value")
-
-	if req.headers.Get("X-New-Header2") != "Value" {
-		t.Errorf("Failed !")
-	}
+	head := req.Headers()
+	assert.Equal(t, "Value", head.Get("X-New-Header2"))
 
 	req.AddStandardHeaders(5)
-
 }
 
 func TestRequestAutoCompleteHeaders(t *testing.T) {
 	req := generateBaseRequestForTesting()
 
 	req.SetAutoCompleteHeaders(true)
 
-	if req.WithAutoCompleteHeaders() == true {
-		t.Logf("Success !")
-	} else {
-		t.Errorf("Failed !")
-	}
+	assert.True(t, req.WithAutoCompleteHeaders(), "Set Autocomplete headers error ")
 }
 
 func TestRequestData(t *testing.T) {
 	req := generateBaseRequestForTesting()
 
 	err := req.SetData([]byte("This is the data now"))
 
-	if err != nil && !bytes.Equal(req.Data(), []byte("This is the data now")) {
-		t.Errorf("Failed !")
-	}
+	assert.NoError(t, err)
+	assert.Equal(t, []byte("This is the data now"), req.Data(), "failed to set data")
 }
 
 func TestRequestSettingRawDataWhenThereIsData(t *testing.T) {
 	req := generateBaseRequestForTesting()
 
 	err := req.SetRawData([]byte("This is the data now"))
 
-	if err != nil && strings.Contains(err.Error(), "data field is already present in this request") {
-		t.Logf("Success !")
-	} else {
-		t.Errorf("Failed %s !", err.Error())
+	expectedError := errors.New("ftw/http: data field is already present in this request")
+	if assert.Error(t, err) {
```

Comment:
Why the `if`? Just write
```go
assert.Error(t, err)
assert.Equal(t, expectedError, err)
```
Or am I missing something?

Same for the instances below.

Reply:
I think I copied that from the testify readme 🤔 

Anyway, works and looks good without the if. Removed.